janus-idp/backstage-plugins
# feat: add scaffolder action for the scaffolder relation link

### Description
Adds a custom scaffolder action appends the `spec.scaffoldedFrom` field to the `catalog-info.yaml` of the generated catalog entities. Used in conjunction with the [scaffolder relation processor](https://github.com/janus-idp/backstage-plugins/pull/1591) to form relation links between scaffolder templates and the entities they generate.

### What issue(s) does this fix?
Fixes: [RHIDP-2221](https://issues.redhat.com/browse/RHIDP-2221)

## How to test changes / Special notes to the reviewer

1. Install the module using the instructions in the README.md
2. Import a template with the scaffolder action. ex: 
  ```yaml
  catalog:
    locations:
      - type: url
        target: https://github.com/Zaperex/backstage-plugins/blob/76062120bd68f78d658f7b4c781611c2c3c95b42/plugins/scaffolder-backend-module-scaffolder-relation/examples/01-scaffolder-relation.yaml
        rules:
          - allow: [Template]
   ```
3. Use the template to register a catalog entity into the catalog and observe that the `spec.scaffoldedFrom` field exists and its value matches the entityRef of the template that generated it.
4. If the [scaffolder relation processor](https://github.com/janus-idp/backstage-plugins/pull/1591), the dependency graph of the ingested component should include a `scaffoldedFrom` link to the template
